specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2017-11-11-preview/searchindex.json

```diff
@@ -64,6 +64,325 @@
+          {
+            "name":"$top",
+            "in": "query",
+            "type": "integer",
+            "format": "int32",
+            "description": "The number of search results to retrieve. This can be used in conjunction with Skip to implement client-side paging of search results. If results are truncated due to server-side paging, the response will include a continuation token that can be passed to ContinueSearch to retrieve the next page of results. See DocumentSearchResponse.ContinuationToken for more information.",
+            "x-ms-client-name": "Top",
+            "x-ms-parameter-grouping": {
+              "name": "SearchParameters"
+            }
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ClientRequestIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ClientAcceptParameter"
+          }
+        ],
+        "x-ms-request-id": "request-id",
+        "responses": {
+          "200": {
+            "description": "Response containing documents that match the search criteria.",
+            "schema": {
+              "$ref": "#/definitions/DocumentSearchResult"
+            }
+          }
+        }
+      }
+    },
+    "/docs/search": {
+      "post": {
+        "tags": [
+          "Search"
+        ],
+        "operationId": "DocumentsProxy_SearchPost",
+        "x-ms-examples": {
+          "SearchIndexPostSearch": { "$ref": "./examples/SearchIndexPostSearch.json" }
+        },        
+        "description": "Searches for documents in the Azure Search index.",
+        "parameters": [
+          {
+            "name": "SearchRequest",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SearchParametersPayload",
+              "description": "The Search request."
+            },
+            "description": "The definition of the Search request."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ClientRequestIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ClientAcceptParameter"
+          }
+        ],
+        "x-ms-request-id": "request-id",
+        "responses": {
+          "200": {
+            "description": "Response containing documents that match the search criteria.",
+            "schema": {
+              "$ref": "#/definitions/DocumentSearchResult"
+            }
+          }
+        }
+      }
+    },
+    "/docs/index": {
+      "post": {
+        "tags": [
+          "Document Index"
+        ],
+        "operationId": "DocumentsProxy_DocumentIndexPost",
+        "x-ms-examples": {
+          "SearchIndexPostIndex": { "$ref": "./examples/SearchIndexPostIndex.json" }
+        },        
+        "description": "Sends a batch of upload, merge, and/or delete actions to the Azure Search index.",
```

Comment:
What about mergeOrUpload?

Reply:
Added. Thanks. 

---
In reply to: [208103487](https://github.com/Azure/azure-rest-api-specs/pull/3578#discussion_r208103487) [](ancestors = 208103487)